rfc/rfc-80/rfc-80.md

```diff
@@ -0,0 +1,187 @@
+
+Read performance:  
+Since the data is already sorted when it is written, the SortMerge method can be used directly to merge the data; compared with the native bucket data reading performance is improved a lot, and the memory consumption is reduced significantly.  
+
+Compaction performance:  
+The logic of compaction and reading is the same. Compaction costs across column families is where there real savings are.  
+
+The log merge we can make it pluggable to decide between hash or sort merge - we need to introduce new log headers or standard mechanism for merging to determine if base file or log files are sorted.
+
+## Background
+Currently, Hudi organizes data according to fileGroup granularity. The fileGroup is further divided into column clusters to introduce the columnFamily concept.  
+The organizational form of Hudi files is divided according to the following rules:  
+The data in the partition is divided into buckets according to hash (each bucket maps to a file group); the files in each bucket are divided according to columnFamily; multiple colFamily files in the bucket form a completed fileGroup; when there is only one columnFamily, it degenerates into the native Hudi bucket table.
+
+![table](table.png)
+
+## Implementation
+This feature should be implemented for both Spark and Flink. So, a table written by Flink this way, also can be read by Spark.
+
+### Constraints and Restrictions
+1. The overall design relies on the non-blocking concurrent writing feature of Hudi 1.0.  
+2. Lower version Hudi cannot read and write column family tables.  
+3. Only MOR bucketed tables support setting column families.  
+   MOR+Bucket is more suitable because it has higher write performance, but this does not mean that column family is incompatible with other indexes and cow tables.  
+4. Column families do not support repartitioning and renaming.  
+5. Schema evolution does not take effect on the current column family table.  
+   Not supporting Schema evolution does not mean users can not add/delete columns in their table, they just need to do it explicitly.
+6. Like native bucket tables, clustering operations are not supported.
+
+### Model change
+After the column family is introduced, the storage structure of the entire Hudi bucket table changes:
+
+![bucket](bucket.png)
+
+The bucket is divided into multiple columnFamilies by column cluster. When columnFamily is 1, it will automatically degenerate into the native bucket table.
+
+![file-group](file-group.png)
+
+### Proposed Storage Format Changes
+After splitting the fileGroup by columnFamily, the naming rules for base files and log files change. We add the cfName suffix at the end of all file names to facilitate Hudi itself to distinguish column families. If it's not present, we assume default column family.
+So, new file name templates will be as follows:  
+- Base file: [file_id]\_[write_token]\_[begin_time]**[#cfName]**.[extension]  
```

Comment:
I guess the `cfName` is separated with a `_` prefix right?

Reply:
Honestly, I wanted to use some extra symbol for separation column family names (to ease their's parsing). But if you prefer using only `_` as a separator in file names - ok, i'll fix it.

Reply:
What is the special symble then, `**` is ambugious because it is a special functional symble in markdown.

Reply:
I mean `#`

Reply:
Fixed separator to `_`.